docs: update the no-required-schema-properties-undefined rule description (#1446)
Redocly/redocly-cli: docs/rules/no-required-schema-properties-undefined.md

  Pet:
    type: object
    required:
      - id
      - name
    properties:
      id:
        type: integer
        format: int64
      name:
        type: string
        example: doggie
```

> The rule works case-sensitive, which means a property `name` will be invalid in combination with the string `Name` in the `required` list:

Reviewer: ```suggestion
The rule is case-sensitive, which means a property `name` does not match the string `Name` in the `required` list:
```
